groups/artifactory/locals.tf

```diff
@@ -0,0 +1,28 @@
+locals {
+  secrets                                = data.vault_generic_secret.secrets.data
+  artifactory_access_token               = local.secrets.artifactory_access_token
+  server_url                             = "${var.service}.${var.environment}.${data.aws_route53_zone.selected.name}:8081/artifactory"
```

Comment:
This is missing the URL schema and so is invalid. Additionally, we're referencing a data resource to interpolate the domain when we already have a suitable local to use - `dns_zone_name`

So this needs to be updated as:
E.g.:
```
server_url                             = "http://${var.service}.${var.environment}.${local.secrets.dns_zone_name}:8081/artifactory"
```